src/utils/promisifyTypeMethod.js

```diff
@@ -0,0 +1,41 @@
+import { getIsArgumentCallback } from './getIsArgumentCallback';
+
+/**
+ * Maps type of argument from scheme to method which return the argument value for original method
+ */
+const schemeArgTypeToExternalArgValueMap = {
+  value: externalArgs => externalArgs.shift(),
+  successCallback: (_, resolveCallback) => resolveCallback,
+  errorCallback: (_, __, rejectCallback) => rejectCallback,
+  objectWithCallback: (externalArg, resolveCallback, rejectCallback) => Object.assign(externalArg || {}, { onSuccess: resolveCallback, onFailure: rejectCallback })
+};
+
+/**
+ * Wraps a method of object in Promise if it is asyncronous method by scheme
+ *
+ * @param {import('../../types').ObjectValue} objValue object whose method must be promised if it needed
+ * @param {string} fieldName method name
+ * @param {import('../../types').SchemeArg[]} argsFromScheme scheme of method arguments
+ * @param {import('../../types').Callback} [mapAfterCallback] mapAfter callback from scheme
+ * @returns {import('../../types').ObjectValue}
+ */
+export function promisifyTypeMethod(objValue, fieldName, argsFromScheme, mapAfterCallback) {
+  const originalMethod = objValue[fieldName];
+  const shouldPromised = argsFromScheme.some(getIsArgumentCallback);
+  if (shouldPromised) {
+    objValue[fieldName] = (...args) => {
+      const resultObject = {};
+      resultObject.callResult = new Promise((resolve, reject) => {
```

Comment:
Can we return promise here, not object? 
```js
return new Promise(...);
```
Why do we need callResult?

Reply:
A lot of methods is really return undefined, but 'webOS.service.request' is return a object which include a fields and methods (cancel for example) and should return this object, because developers can use it.

Reply:
Maybe i should rename callReturn and callResult because understanding of it is really difficultly